src/screens/ProfileScreen/elements/TextWithEmoji.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import {Text, StyleSheet} from 'react-native';
+import emojiRegex from 'emoji-regex';
+import {COLORS} from '../../../utils/theme';
+
+export function TextWithEmoji({text}: {text: string}) {
+  return (
+    <Text style={styles.text}>
+      {text.split(' ').map((item: string, index: number) => {
+        return (
+          <Text key={index} style={item.match(emojiRegex()) ? styles.emoji : styles.text}>
+            {`${item} `}
+          </Text>
+        );
+      })}
+    </Text>
+  );
+}
+
+const styles = StyleSheet.create({
+  text: {
+    color: COLORS.lightgrey,
+    fontFamily: 'Inter',
+    fontSize: 14,
+    fontStyle: 'normal',
+    fontWeight: '400',
+    marginBottom: 12,
+    lineHeight: 20
+  },
+  emoji: {
+    fontSize: 14,
+    fontFamily: 'System'
+  }
+});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `TextWithEmoji` component seems to be well implemented. It correctly uses the `emoji-regex` library to detect emojis and apply different styles to them. However, there is a potential performance issue with the current implementation. The `text.split(' ').map()` operation is performed every time the component re-renders, which could be costly if the text is long. To optimize this, you can use the `useMemo` hook from React to only perform this operation when the `text` prop changes.

Here's how you can do it:

```diff
export function TextWithEmoji({text}: {text: string}) {
+  const words = React.useMemo(() => text.split(' '), [text]);
  return (
    <Text style={styles.text}>
-       {text.split(' ').map((item: string, index: number) => {
+       {words.map((item: string, index: number) => {
        return (
          <Text key={index} style={item.match(emojiRegex()) ? styles.emoji : styles.text}>
            {`${item} `}
          </Text>
        );
      })}
    </Text>
  );
}
```

Also, using the index as a key in React lists is not recommended because it can lead to unexpected behavior during re-renders if items are added, removed, or reordered. If the `text` prop is guaranteed to be unique, you could use that as the key instead.

```diff
- <Text key={index} style={item.match(emojiRegex()) ? styles.emoji : styles.text}>
+ <Text key={`${item}-${index}`} style={item.match(emojiRegex()) ? styles.emoji : styles.text}>
```
This way, even if the order of words changes, each word will still have a unique key.


<!-- This is an auto-generated comment by OSS CodeRabbit -->